README.md

```diff
@@ -1,32 +1,127 @@
+
+- **Initialization**:  Gradle locates the build files it must process and
+  determines whether the build is single-project or multi-project. The build files
+  are passed to the next phase.
+- **Configuration**: Gradle executes each build file as a Groovy or Kotlin script
+  to create a directed acyclic graph (DAG) of task objects.
+  - “Directed” means each dependency arrow goes in one direction. 
+  - “Acyclic” means that there are no loops in the graph.
+- **Execution**: Gradle identifies the tasks in the task DAG  
+  and executes them in dependency order to ensure each task is executed only once.
+
+For example, the following image shows the DAG for a Java build:
+
+![java plugin DAG](https://curriculum-content.s3.amazonaws.com/6002/gradle-lifecycle/java-dag.png)
+
+Each task defines a set of inputs and outputs.
+The inputs for **compileJava** are the Java source files, while the
+outputs are class files. If none of the inputs or outputs
+have changed, Gradle will skip that task. 
+We call this Gradle’s **incremental build support**.
+
+Let's explore some of this using IntelliJ's Gradle Tool Window,
+which can be opened either by clicking on "Gradle" along
+the edge of the IntelliJ window, or by selecting ** View | Tool Windows | Gradle**
+from the main menu.
+  
+![gradle tool window](https://curriculum-content.s3.amazonaws.com/6002/gradle-lifecycle/gradletoolwindow.png)
+
+### Tasks
+
+Expand the **Tasks** folders to see tasks available for execution.
+
+![gradle tasks](https://curriculum-content.s3.amazonaws.com/6002/gradle-lifecycle/gradletasks.png)
+
+- Gradle's Base plugin provides general lifecycle tasks such as *clean*,
+  *assemble*, and *build*.
+- The Java plugin provides additional tasks related to compiling and
+  packaging a Java project: *compileJava*, *processResources*, *classes*,
+  *compileTestJava*, *processTestResources*, *testClasses*, *jar*, *javadoc*,
+  *test*, and *cleanTaskName*.
+
+We can get a description of each task by typing "gradle tasks --all"
+in the terminal window.
+
+## Build a project
+
+Let's see how IntelliJ builds the project when we change the code
+and need to recompile. 
+
+Edit the `main` method to print "Hello" instead of "Hello World!", then:
+
+1. Press the build icon on the bottom toolbar to display the build view.
+2. Press the build icon on the top toolbar or
+   select **Build | Build Project** from the main menu to build the project.  
+
+![build](https://curriculum-content.s3.amazonaws.com/6002/gradle-lifecycle/gradlebuild.png)
+
+Building from the main menu executes the **classes**
+and **testClasses** tasks, along with other dependent tasks
+based on the DAG. Tasks that are up-to-date or have no inputs are skipped.
+
+What if we want to execute the **build** task itself?
+There are two different ways to do this:
+
+1. Right-click on the task in the Gradle Tool Window and select "Run":
+   ![run build](https://curriculum-content.s3.amazonaws.com/6002/gradle-lifecycle/runbuild.png)
```

Comment:
Might need some new lines in between the images and text as it looks like the formatting is off (at least in GitHub).

Reply:
converted to a table